packages/react-hooks/src/useMergedRefs.ts

```diff
@@ -1,24 +1,21 @@
-import { useRef, useCallback, Ref, MutableRefObject } from 'react';
-
+import { useCallback, Ref, MutableRefObject } from 'react';
 /**
  * React hook to merge multiple React refs (either MutableRefObjects or ref callbacks) into a single ref callback that
  * updates all provided refs
  * @param refs- Refs to collectively update with one ref value.
  */
 export function useMergedRefs<T>(...refs: Ref<T>[]): (instance: T) => void {
-  const state = useRef<(Ref<T> | undefined)[]>();
-
-  // Update refs list.
-  state.current = refs;
-
-  return useCallback((value: T) => {
-    for (const ref of state.current!) {
-      if (typeof ref === 'function') {
-        ref(value);
-      } else if (ref) {
-        // work around the immutability of the React.Ref type
-        ((ref as unknown) as MutableRefObject<T>).current = value;
+  return useCallback(
+    (value: T) => {
+      for (const ref of refs) {
+        if (typeof ref === 'function') {
+          ref(value);
+        } else if (ref) {
+          // work around the immutability of the React.Ref type
+          ((ref as unknown) as MutableRefObject<T>).current = value;
+        }
       }
-    }
-  }, []);
+    },
+    [...refs],
```

Comment:
This breaks rules of hooks, https://github.com/facebook/react/blob/fed4ae0247883050aab6d77d9a83f9b6399d9a45/packages/eslint-plugin-react-hooks/src/ExhaustiveDeps.js#L737

Reply:
It may violate the ESLint rule logic, but it doesn't violate the intention or spirit of the rule. The alternative is either having N overloads handling N refs, or forcing users to chain `useMergedRefs()`